### ISSUE TITLE
Incorrect duplicate identifier error in JS files when property assignments and accessors have same name

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#10280]
```js
class C {
    constructor(thing) {
        this.thing = thing;
    }
    get thing() {
        return this._thing;
    }
    set thing(value) {
        this._thing = value;
    }
}
```

Expected: No errors.
Actual: Duplicate identifier errors on both accessors and the assignment to `this.thing`. Really, the constructor is *taking advantage* of the presence of the accessors.

This happened when I tried to recompile our JS output of RxJS.

**Comment by mhegazy** [comment]
working for me correctly on master. 

```cmd
C:\Tools>tsc --allowjs C:\test\sandbox8\a.js --noemit

C:\Tools>type c:\test\sandbox8\a.js
class C {
    constructor(thing) {
        this.thing = thing;
    }
    get thing() {
        return this._thing;
    }
    set thing(value) {
        this._thing = value;
    }
}
C:\Tools>tsc --allowjs C:\test\sandbox8\a.js --noemit

C:\Tools>echo %ERRORLEVEL%
0
```
